tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -295,20 +295,141 @@ INSTANTIATE_TEST_SUITE_P(
     Reduce,
     PipelineTestTwoStages,
     testing::Combine(
-        all_backends,
+        testing::Values(CommunicatorBackend::nccl),
         all_nontrivial_meshes,
         all_meshes,
         testing::Values(true),
         testing::Values(false),
         testing::Values(true),
         testing::Values(0, 1),
-        testing::Bool()));
+        testing::Values(false)));
 
 INSTANTIATE_TEST_SUITE_P(
     ReduceScatter,
     PipelineTestTwoStages,
     testing::Combine(
-        all_backends,
+        testing::Values(CommunicatorBackend::nccl),
+        all_nontrivial_meshes,
+        testing::Values(mesh_null), // the same mesh is used for all tensors
+        testing::Values(true),
+        testing::Values(true),
+        testing::Values(true),
+        testing::Values(0, 1),
+        testing::Values(false)));
+
+// TODO: Distributed reduction tests using fusion executor cache are failing
+// AllocationDomainPass might be re-ordering compute
```

Comment:
We can disable the optimization pass here like here. This is probably the easiest approach
https://github.com/NVIDIA/Fuser/blob/cf63eb6b6cc1d91aa6bdbfb0e3ac729ecebb5e58/tests/cpp/test_matmul_scheduler.cpp#L28

Reply:
I think a *cleaner* approach is to specify allocation domain on output. Alternatively, if we think any computation coming in with a device mesh is too sensitive to be touched by optimization passes, I should probably also skip those in my optimization passes.

I think similar thing would apply to alias analysis as well. wondering if @wujingyue 's pass would do the same.

Reply:
Thanks for the pointers! I imagine for a while the multidevice tests will be sensitive to many optimizations until we are more stable. It might be better for us to disable optimization passes on the multidevice test. Later we can make the pass composable with multidevice. 

I think @naoyam mentioned he is setting the allocation domain for all tensors as part of index refactoring. Ideally, this will help make things more robust.

Reply:
> I think @naoyam mentioned he is setting the allocation domain for all tensors as part of index refactoring. Ideally, this will help make things more robust.

Yeah that's indeed useful. Though I'm imagining that is going to happen after the initial optimization pass... Conceptually I think it makes sense (at least for allocation order pass) to avoid touching fusion related to multi device. Given that we know such constraints applied to tensors involved with multidevice at the time. I'll try to patch that.

Reply:
> I think a cleaner approach is to specify allocation domain on output. 

Correct. I think [insertShardingAxisReordering](https://github.com/NVIDIA/Fuser/blob/cf63eb6b6cc1d91aa6bdbfb0e3ac729ecebb5e58/csrc/multidevice/utils.cpp#L257) should set allocation domain explicitly. An empty allocation domain means a wildcard. 

> I think it makes sense (at least for allocation order pass) to avoid touching fusion related to multi device. 

That should yield to the correct solution, IMO. AllocationOrderPass and MarkAliasesPrepare are and will be still useful for non-DIDs. Wdyt?

Reply:
Yes I think these passes are still useful. Setting the allocation domain insertShardingAxisReordering is a great suggestion. I'll find some time to add this.